service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredTaskExecutor.java

```diff
@@ -79,8 +90,26 @@ public void execute(GetGpcStructuredTaskDefinition structuredTaskDefinition) {
             messageContext.resetMessageContext();
         }
 
+        var payload = hl7TranslatedResponse;
```

Comment:
This will make a copy of a quite large string in memory. Reuse the `hl7TranslatedResponse` variable